New extension: Extendable Blocks

CST1229 opened the pull request:
Blocks that extend, like in Snap<i>!</i>.
![image](https://github.com/TurboWarp/extensions/assets/68464103/7408d792-5ddc-4bdb-ad24-34fc086c10cb)


This uses a ton of scratch-blocks patches and also may be considered a utilities extension? (Though splitting this would require duplicating a bunch of code so...) So I don't know if this will be accepted or not.

CST1229 committed:
extendable blocks

CST1229 committed:
fix create json array block in compiler

CST1229 committed:
idk

CST1229 committed:
format

CST1229 committed:
remove patch functions

CST1229 committed:
format again

CST1229 committed:
some more comments

CST1229 committed:
svg width/height

CST1229 committed:
throw an error instead of alert()ing

LilyMakesThings commented:
I wouldn't consider this a Utilities extension because there is a consistent theme between each block - it also makes more practical sense to include these blocks into one extension because of the patches you're doing to ScratchBlocks.

fo-od commented:
i think having + and - icons would make more sense since you adding more, but it also makes sense to have arrows because the extension is calling extendable blocks

VoxelDoesCode commented:
The two arrows feel more in line with Scratch's design language, in my opinion.

LilyMakesThings commented:
The arrows came from the dropdown menus so they match the style, too.

Arrows are also more consistent with other familiar mods, such as Snap_!_

DilemmaGX commented:
![image](https://github.com/TurboWarp/extensions/assets/111679857/72ca64e1-09c1-4556-8f70-0075ce80673d)
I don't know why, but this extension cannot run on some versions of Turbowarp Desktop (for example, in this case, `v1.9.3`). Anyway, this extension is great running in turbowarp.org.

CST1229 commented:
> I don't know why, but this extension cannot run on some versions of Turbowarp Desktop (for example, in this case, `v1.9.3`).

Use the latest 1.10 beta; this extension requires some things that were only added in the merge-upstream update (the update which added color theme options and a bunch of addons).

Gen1x-ALT commented:
this looks based

CST1229 committed:
Merge branch 'master' of https://github.com/TurboWarp/extensions into extendable-blocks

CST1229 committed:
license

CST1229 committed:
lint

CST1229 committed:
Make everything l10n-able
see CST1229/turbowarp-extensions#1

CST1229 committed:
remove "you" on some of the descriptions

CST1229 committed:
more string changes

DilemmaGX commented:
Hello, may I embed it into [GekoWarp](https://dilemmagx.github.io/GekoWarp)'s extension library?

CST1229 committed:
Merge branch 'master' of https://github.com/TurboWarp/extensions into extendable-blocks

CST1229 committed:
misc changes
- tweak description
- remove MAX_EXTENDABLE_INPUTS (it's been a while)
- some code cleanup

CST1229 committed:
Merge branch 'master' of https://github.com/TurboWarp/extensions into extendable-blocks

PPPDUD commented:
I haven't taken an in-depth review yet, but it looks interesting.

CST1229 committed:
what was going on here

LSPECTRONIZTAR commented:
one prob: the json block will return 2 items even if there are more than 2 inputs

CST1229 committed:
cast boolean inputs to boolean

yuri-kiss reviewed (changes requested):
I haven't looked at the code or anything nor have I tested this, I am just making this review to hopefully stop someone from merging this to early before we have confirmation on these... patchy type extensions.

CubesterYT commented:
> I haven't looked at the code or anything nor have I tested this, I am just making this review to hopefully stop someone from merging this to early before we have confirmation on these... patchy type extensions.

Doubt ANYONE feels ok merging patch type extensions 😬
